internal/docker/builder.go

```diff
@@ -260,6 +260,18 @@ func (d *Builder) construct(bprint b.Blueprint) (errs []error) {
 		}
 		// kill the container
 		defer func(r result) {
+			containerInfo, err := d.Docker.ContainerInspect(context.Background(), r.containerID)
+
+			if err != nil {
+				d.log("%s : Can't get status of %s", r.contextStr, r.containerID)
+				return
+			}
+
+			if !containerInfo.State.Running {
+				// The container isn't running anyway, so no need to kill it.
```

Comment:
We are now stopping the containers so I don't see how `containerInfo.State.Running` can ever be `true`?

Reply:
I don't have a great deal of Go experience, but I view this defer as a last-resort clean-up mechanism, like a finally block in Java/etc. Anecdotally, I've had docker stop leave a container running before, but frankly this may well have been a bug (I'm not sure). I'm happy to remove it if you think that's best, but it really was just intended as a last resort 'stopping gracefully is nice, but littering the host with containers is to be avoided if at all possible'.

Reply:
If you've seen this in the wild then let's leave it in. Docker is async command wise (you'll note several places we repeat the same instruction until timeout/success) so I can believe it not stopping the container when you ask for it to stop.